Userland/Libraries/LibCrypto/Hash/SHA1.cpp

```diff
@@ -67,6 +81,63 @@ inline void SHA1::transform(u8 const* data)
     secure_zero(blocks, 16 * sizeof(u32));
 }
 
+#if (ARCH(I386) || ARCH(X86_64)) && defined(AK_COMPILER_GCC)
+
+SHA1_ATTRIBUTE_TARGET_X86 static void transform_impl(u32 (&state)[5], u8 const (&data)[64])
+{
+    using AK::SIMD::i32x4;
+
+    i32x4 abcd[2] {};
+    i32x4 msgs[4] {};
+
+    abcd[0] = AK::SIMD::load_unaligned<i32x4>(&state[0]);
+    abcd[0] = AK::SIMD::item_reverse(abcd[0]);
+    i32x4 e { 0, 0, 0, static_cast<i32>(state[4]) };
+
+    // FIXME: Test if unrolling the loop is worth it
+    //          GCC: #pragma GCC unroll(5)
+    //        Clang: #pragma unroll
+#    define GROUP(i_group)                                                                         \
```

Comment:
Why can't this be lambda?

Reply:
Because the `sha1rnds4` instruction takes an `imm8` not variable or pointer or register.

Reply:
Then `i_group` should be a NTTP:
```cpp
auto group = [&]<int i_group>() {
```

Reply:
OK, will fix.